mix/sphinx.nim

```diff
@@ -1,10 +1,12 @@
 import results, sequtils
+import nimcrypto/sysrand
 import ./[config, crypto, curve25519, serialization, tag_manager]
 
 # Define possible outcomes of processing a Sphinx packet
 type ProcessingStatus* = enum
   Exit # Packet processed successfully at exit
   Intermediate # Packet processed successfully at intermediate node
+  Reply # Packet processed successfully at exit; a reply message
```

Comment:
this is semi-unclear: is it a reply message processed at the exit of the _surb_ destination (i.e. where it all started), or is this a packet that has been processed at the exit, same as `Exit`, but in addition to that, it also has a surb, so is a message that expects a reply?